tests/jballerina-unit-test/src/test/resources/test-src/expressions/varref/record-variable-reference.bal

```diff
@@ -369,6 +369,74 @@ function testMappingBindingWithSingleNameFieldBinding() {
     assertEquality("bar4", b);
 }
 
+function testMappingBindingPatternAgainstOpenRecordInTupleDestructuring() {
+    [Some, Some] r1 = [{var1: "A", var2: "B"}, {var1: "C", var2: "D"}];
+    string xVar1;
+    string xVar2;
+    [{var1: xVar1}, {var1: xVar2}] = r1;
+    assertEquality("A", xVar1);
+    assertEquality("C", xVar2);
+
+    [record {int a; int b;}] r2 = [{a: 1, b: 2}];
+    int a;
+    int b;
+    [{a, b}] = r2;
+    assertEquality(1, a);
+    assertEquality(2, b);
```

Comment:
In addition to this, let's add one where the value has rest fields.

Reply:
Were these added?

Reply:
I have added L387-L390

Reply:
Can you point to the updated line numbers?

Reply:
L355..L371

Reply:
Not exactly what I suggested? Please add a simple test like this with rest fields.

Reply:
Added. L543...L546

Reply:
What I meant was like just

```ballerina
[record {int a; int b;}] r2 = [{a: 1, b: 2, "c": "three", "d": 4}];
```

Reply:
That's a spread field in mapping constructors - https://ballerina.io/spec/lang/master/#spread-field